code-scanning/policy-validator-tf.yaml

```
@@ -0,0 +1,117 @@
+# 2. Copy this workflow and add it under .github/workflows folder of your GitHub repository with name `policy-validator-iam-policies-tf.yaml`
+# 3. If you're using CheckNoNewAccess check, you need to create a reference policy. Use the guide [here](https://github.com/aws-samples/iam-access-analyzer-custom-policy-check-samples?tab=readme-ov-file#how-do-i-write-my-own-reference-policies)and store them in S3 bucket / GitHub secrets to compare them against the policies in the TF templates. In below workflow, we are storing the reference policy in the GitHub secrets with name `REFERENCE_IDENTITY_POLICY`
+# 4. If you're using the CheckAccessNotGranted check, identify the critical actions that shouldn't be granted access by the policies in the TF templates. Store these actions in S3 bucket / GitHub secrets to compare them against the policies in the TF templates. In the below workflow, we are storing the S3 bucket object containing the critical action in the GitHub secret with name `CRITICAL_ACTIONS`
+# 5. Create a new workflow under ./github/workflows and refer this workflow. Configure the workflow with events to run and the path to the terraform plan to be validated. Reference sample code:
+# ```
+# name: Policy Validator for AWS IAM policies in Terraform templates
+# on:
+#   pull_request:
+#     types: [opened, review_requested]
+#   push:
+#     branches:
+#       - 'main'
+# jobs:
+#     security-scanner-developers:
+#         uses: .github/workflows/policy-validator-iam-policies-tf.yaml@main
+#         secrets: inherit
+#         with:
+#             template-path: file-path-to-tf-plan.yaml #Path to the terraform plan
+#             region: us-west-2
+# ```
+# 4. Start using the GitHub actions by generating the GitHub events matching the defined criteria in your workflow.
+
+name: Re-usable workflow for validating AWS IAM policies in Terraform templates using Policy Validator
+on:
+  # This workflow is written to illustrate it to use it as re-usable workflow.
+  workflow_call:
+    inputs:
+      template-path:
+        required: true
+        type: string
+      region:
+        required: true
+        type: string
+    outputs:
+      result:
+        value: string
+  # Uncomment the below lines if you want to run this workflow as against push / pull request of default branch
+  # Note: Please pass the value of the inputs in the below workflow by replacing ${{ inputs.<variable> }} with value if you don't want to use it as a re-usable workflow
+  # push:
+  #   branches: [$default-branch, $protected-branches]
+  # pull_request:
+  #   # The branches below must be a subset of the branches above
+  #   branches: [$default-branch]
```

Similar to the prior note, uncommenting by default to match the style guide:

```suggestion
  # Note: Please pass the value of the inputs in the below workflow by replacing ${{ inputs.<variable> }} with value if you don't want to use it as a re-usable workflow
  push:
    branches: [$default-branch, $protected-branches]
  pull_request:
    # The branches below must be a subset of the branches above
    branches: [$default-branch]
```